discord/automod.py

```diff
@@ -176,36 +179,71 @@ class AutoModTriggerMetadata:
     Attributes
     ----------
     keyword_filter: List[:class:`str`]
-        A list of substrings to filter. Only for triggers of type :attr:`AutoModTriggerType.keyword`.
+        A list of substrings to filter.
+        Only for triggers of type :attr:`AutoModTriggerType.keyword`.
+    regex_patterns: List[:class:`str`]
+        A list of regex patterns to filter using Rust-flavored regex, which is not
+        fully compatible with regex syntax supported by the builtin `re` module.
+        Only for triggers of type :attr:`AutoModTriggerType.keyword`.
+
+        .. versionadded:: 2.4
     presets: List[:class:`AutoModKeywordPresetType`]
-        A list of keyword presets to filter. Only for triggers of type :attr:`AutoModTriggerType.keyword_preset`.
+        A list of keyword presets to filter.
+        Only for triggers of type :attr:`AutoModTriggerType.keyword_preset`.
+    allow_list: List[:class:`str`]
+        A list of substrings to allow, overriding keyword and regex matches.
+        Only for triggers of type :attr:`AutoModTriggerType.keyword` and :attr:`AutoModTriggerType.keyword_preset`.
+
+        .. versionadded:: 2.4
+    mention_total_limit: :class:`int`
```

Comment:
Should be noted that this is only valid for the mention trigger type, similar to above